shared/src/types/storage.rs

```diff
@@ -236,6 +240,19 @@ impl Key {
             _ => false,
         }
     }
+
+    /// Check if the given key is a key to IBC-related data
+    pub fn is_ibc_key(&self) -> bool {
+        self.segments[0] == DbKeySeg::StringSeg(RESERVED_IBC_KEY.to_owned())
```

Comment:
the access by index will panic for an empty vec - it would be safer to use `get`